widget/entry.go

```diff
@@ -869,11 +941,19 @@ func (e *Entry) eraseSelection() {
 		return
 	}
 
+	erasedText := provider.String()[posA:posB]
```

Comment:
You are building and allocating an entire new copy of the string contents from all richtext segments here. Can you not use the `entry.Text` field in this case?

Reply:
But the code you are referencing is called after the text has changed? What I am referring to is called when I believe them to be in sync

Reply:
I just made the changes.